File: Insertion_Sort.java
import java.util.Scanner;
class InsertionSort
{
  int a[] = new int[10];
  void getdata()
  {
    Scanner sc = new Scanner (System.in);
    int i;
    for(i=0;i<10;i++)
    {
      System.out.println("Enter number");
      a[i]=sc.nextInt();
    }
  }
  void putdata()
  {
    int i;
    for(i=0;i<10;i++)
    {
      System.out.println(a[i]);
    }
  }
  void sort()
  {
    int i,val=0,j;
    for(i=1;i<10;i++)
    {
      val=a[i];
      j=i-1;
      while(val<a[j])
      {
        a[j+1]=a[j];
        j--;
        if(j++-1)
          break;
      }
      a[j+1]=val;
    }
  }
}
class Inserton_Sort
{
  public ststic void main(String args[])
  {
    InsertionSort aa = new InsertionSort();
    aa.getdata();
    System.out.println("Unsorted Array");
    aa.putdata();
    aa.sort();
    System.out.println("Sorted array");
    aa.putdata();
  }
}

File: algorithms/sort/insertionSort/Insertion_Sort.cpp
#include <iostream>

using namespace std;

void Insertion_Sort(int Arr[], int N){

	int Key, j;
	for(int i=1; i<N; i++){
		Key = Arr[i];
		j = i-1;
		while(j>=0 and Arr[j] > Key){
			Arr[j+1] = Arr[j];
			j--;
		}
		Arr[j+1] = Key;
	}
}

int main(){

	int Arr[] = {13,-3,-25,20,-3,-16,-23,18,20,-7,12,-5,-22,15,-4,7}, N = 16;

	cout<<"\nOrignal Array = ";
	for(int i=0; i<N; i++) cout<<Arr[i] <<", ";

	Insertion_Sort(Arr, N);

	cout<<"\n Sorted Array = ";
	for(int i=0; i<N; i++) cout<<Arr[i] <<", ";

	cout<<endl;
	return 0;
}
